scripts/Crawlers/ZenodoCrawler.py

```diff
@@ -0,0 +1,399 @@
+    with open(dats_path, "w") as f:
+        data = {
+            "title": dataset["original_title"],
+            "identifier": dataset["identifier"],
+            "creators": dataset["creators"],
+            "description": dataset["description"],
+            "version": dataset["version"],
+            "licenses": dataset["licenses"],
+            "keywords": dataset["keywords"],
+            "distributions": dataset["distributions"],
+            "extraProperties": dataset["extraProperties"],
+        }
+
+        # Count number of files in dataset
+        num = 0
+        for file in os.listdir(dataset_dir):
+            if file[0] == "." or file == "DATS.json" or file == "README.md":
+                continue
+            elif os.path.isdir(file):
+                num += sum(
+                    [
+                        len(list(filter(lambda x: x[0] != ".", files)))
+                        for r, d, files in os.walk(file)
+                    ]
+                )
+            else:
+                num += 1
+        data["extraProperties"].append(
+            {"category": "files", "values": [{"value": str(num)}]}
+        )
+
+        # Retrieve modalities from files
+        file_paths = map(
+            lambda x: x.split(" ")[-1],
+            filter(lambda x: " " in x, Repo(dataset_dir).git.annex("list").split("\n")),
+        )  # Get file paths
+        file_names = list(
+            map(lambda x: x.split("/")[-1] if "/" in x else x, file_paths)
+        )  # Get file names from path
+        modalities = set([_guess_modality(file_name) for file_name in file_names])
+        if len(modalities) == 0:
+            modalities.add("unknown")
+        elif len(modalities) > 1 and "unknown" in modalities:
+            modalities.remove("unknown")
+        data["types"] = [{"value": modality} for modality in modalities]
+
+        json.dump(data, f, indent=4)
+
+
+def _create_zenodo_tracker(path, dataset, private_files, restricted):
+    with open(path, "w") as f:
+        data = {
+            "zenodo": {
+                "concept_doi": dataset["concept_doi"],
+                "version": dataset["latest_version"],
+            },
+            "private_files": private_files,
+            "restricted": restricted,
+            "title": dataset["original_title"],
+        }
+        json.dump(data, f, indent=4)
+
+
+def _create_readme(dataset, dataset_dir):
+    if "README.md" not in os.listdir(dataset_dir):
+        with open(os.path.join(dataset_dir, "README.md"), "w") as f:
+            f.write(
+                """# {0}
+
+[![DOI](https://www.zenodo.org/badge/DOI/{1}.svg)](https://doi.org/{1})
+
+Crawled from Zenodo
+
+## Description
+
+{2}""".format(
+                    dataset["title"],
+                    dataset["doi_badge"],
+                    html2markdown.convert(dataset["description"]).replace(
+                        "\n", "<br />"
+                    ),
+                )
+            )
```

Comment:
Move to base class, call it from run(), and add an abstract function that returns a markdown string with the content of this file.